### PR TITLE
simplify device compiler `cpp` file selection

### PR DESCRIPTION
**Pull request by psychocoderHPC**
Instead of maintaining a list of files which should compiled by the device compiler we select all files ending with `.x.cpp`.

This is required for the future support of multiple compile units. Maintaining the list by hand would be a nightmare for 20+ cpp files in different folders.

### CODE REVIEW

File: include/picongpu/CMakeLists.txt [hunk 1]
```diff
@@ -586,7 +583,7 @@ if(PIC_HAVE_openPMD)
         picongpu-hostonly
         SYSTEM PRIVATE
         $<TARGET_PROPERTY:toml11::toml11,INTERFACE_INCLUDE_DIRECTORIES>)
-    target_link_libraries(picongpu-hostonly PRIVATE toml11::toml11)
+    target_link_libraries(picongpu-hostonly PUBLIC toml11::toml11)
```
Comment (ikbuibui) [inline include/picongpu/CMakeLists.txt]: Why is this changed to ``PUBLIC``?

**Review by ikbuibui — COMMENTED** [review]
Seems like a reasonable idea. x for xcelerator

File: include/picongpu/CMakeLists.txt (hunk 1 above)
Comment (psychocoderHPC) [reply include/picongpu/CMakeLists.txt]: oh this sneaked in from the PR where I cherry-picked it from. I will remove it.

**Comment by psychocoderHPC** [comment]
> Seems like a reasonable idea. x for xcelerator

Yes and it keeps syntax highlighting in all IDE's without any configuration. This is much better than our `.kernel`, `.tpp` we are using very often.

**Review by chillenzer — APPROVED** [review]
Sounds okay.

**Review by ikbuibui — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
failing CI jobs will be fixed with #5044
